src/pages/nft/claim/index.vue

```diff
@@ -0,0 +1,124 @@
+<template>
+  <div class="flex flex-col justify-center flex-grow">
+    <AuthRequiredView
+      class="flex flex-col relative w-full max-w-[962px] mx-auto mb-[48px]"
+      :login-label="$t('nft_claim_login_in', { nftName })"
+      :login-button-label="$t('nft_claim_login_in_button')"
+    >
+      <div
+        v-if="!isLoading && !isClaimed"
+        class="flex flex-col justify-center gap-[16px] mt-[16px]"
+      >
+        <Label :text="$t('nft_claim_claim', { nftName, wallet: getAddress })" align="center" />
+        <ButtonV2 :text="$t('nft_claim_claim_button')" preset="secondary" @click="claim" />
+      </div>
+      <div v-else-if="isLoading">
+        <Label :text="$t('nft_claim_claiming', { nftName })" align="center" />
+        <ProgressIndicator class="self-center mt-[16px]" />
+      </div>
+      <div v-else-if="isClaimed">
+        <Label :text="$t('nft_claim_claimed', { nftName })" align="center" />
+        <ButtonV2 :text="$t('nft_claim_claimed_view_button')" preset="secondary" @click="handleClickView" />
+      </div>
+      <Label v-else-if="error" align="center">
+        {{ $t('nft_claim_error_message', { error }) }}
+      </Label>
+      <Label v-else align="center">
+        {{ $t('nft_claim_error_message_unknown') }}
+      </Label>
+      <div class="flex justify-center gap-[16px] mt-[16px]">
+        <ButtonV2
+          v-if="!isLoading && !isClaimed && error"
+          :text="$t('nft_claim_claimed_retry_button')"
+          preset="outline"
+          @click="handleClickRetry"
+        />
+      </div>
+    </AuthRequiredView>
+  </div>
+</template>
+
+<script>
+import { logTrackerEvent } from '~/util/EventLogger';
+import { postStripeFiatClaim } from '~/util/api';
+
+import alertMixin from '~/mixins/alert';
+import walletMixin from '~/mixins/wallet';
+
+export default {
+  name: 'NFTClaimPage',
+  mixins: [alertMixin, walletMixin],
+  data() {
+    return {
+      classId: '',
+      nftId: '',
+      isLoading: false,
+      error: '',
+    };
+  },
+  computed: {
+    nftName() {
+      const name = this.$route.query.claiming_class_name;
+      return `Writing NFT${name ? ` - ${decodeURIComponent(name)}` : ''}`;
+    },
+    paymentId() {
+      return this.$route.query.payment_id;
+    },
+    token() {
+      return this.$route.query.claiming_token;
+    },
+    isClaimed() {
+      return this.classId && this.nftId;
+    },
+  },
+  methods: {
+    async claim() {
+      try {
+        this.isLoading = true;
+        if (!this.paymentId) {
+          throw new Error('id is required');
+        }
+        if (!this.token) {
+          throw new Error('token is required');
+        }
```

Comment:
We can also automatically show a simple message in UI that tells user to open the link in email (with token and etc) if this page is missing any required qs, no need to wait for claim action